### PR TITLE
Add hex value generation, both for as color and as random, see issue #282

### PR DESCRIPTION
**Pull request by haruntuncay**
Added a way to generate hex string from `faker.random(int length) ` and `faker.color().hex()`, related to issue #282.
**Example usages:**

```
Faker faker = new Faker(); //
System.out.println(faker.random().hex());  // 768AB21E
System.out.println(faker.random().hex(4)); // 820E
System.out.println(faker.color().hex()); // #F78DFE
System.out.println(faker.color().hex(false)); // DC095B No "#" sign.
```

### CODE REVIEW

File: src/main/java/com/github/javafaker/service/RandomService.java [hunk 1]
```diff
@@ -4,8 +4,13 @@
 
 public class RandomService {
     private static final Random SHARED_RANDOM = new Random();
+    private static final char[] hexValues;
     private final Random random;
 
+    static {
+        hexValues = new char[]{'0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'A', 'B', 'C', 'D', 'E', 'F'};
```
Comment (codingricky) [inline src/main/java/com/github/javafaker/service/RandomService.java]: could you just inline this? I don't see the benefit of using a static block here

File: src/test/java/com/github/javafaker/service/RandomServiceTest.java [hunk 2]
```diff
@@ -62,4 +63,9 @@ public void testIntInRange() {
             assertThat(randomService.nextInt(-5, 5), both(lessThanOrEqualTo(5)).and(greaterThanOrEqualTo(-5)));
         }
     }
+
+    @Test
+    public void testHex() {
+        assertThat(faker.random().hex(8), matchesRegularExpression("^[0-9A-F]{8}$"));
```
Comment (codingricky) [inline src/test/java/com/github/javafaker/service/RandomServiceTest.java]: This should be called through the randomservice not the faker

File: src/test/java/com/github/javafaker/ColorTest.java [hunk 3]
```diff
@@ -11,4 +11,14 @@
...
         assertThat(faker.color().name(), matchesRegularExpression("(\\w+ ?){1,2}"));
     }
+
+    @Test
+    public void testHex() {
+        assertThat(faker.color().hex(), matchesRegularExpression("^#[0-9A-F]{6}$"));
+    }
+
+    @Test
+    public void testHexNoHashSign() {
```
Comment (codingricky) [inline src/test/java/com/github/javafaker/ColorTest.java]: technically you should have a test that tests passing in a false... so `testHexWithHashSign()`

**Comment by haruntuncay** [comment]
Hey @codingricky,
Thanks for the review. I have committed a fix that covers the pointed out parts.

**Comment by haruntuncay** [comment]
Hello @codingricky,
Hope you are doing well.
When you have the time, could you please check out this PR and #364.
Thank you in advance.  :)

**Pull request merged**